packages/components/src/internal/components/domainproperties/OntologyLookupOptions.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { PureComponent, ReactNode } from 'react';
+import { Col, FormControl, Row } from 'react-bootstrap';
+import { List } from 'immutable';
+import { getServerContext } from '@labkey/api';
+
+import { DomainField, LabelHelpTip } from '../../..';
+import { helpLinkNode, ONTOLOGY_TOPIC } from '../../util/helpLinks';
+
+import { isFieldFullyLocked } from './propertiesUtil';
+import { createFormInputId, fetchOntologies } from './actions';
+import {
+    DOMAIN_FIELD_ONTOLOGY_IMPORT_COL,
+    DOMAIN_FIELD_ONTOLOGY_LABEL_COL,
+    DOMAIN_FIELD_ONTOLOGY_SOURCE,
+} from './constants';
+import { ITypeDependentProps, OntologyModel } from './models';
+import { SectionHeading } from './SectionHeading';
+
+interface Props extends ITypeDependentProps {
+    field: DomainField;
+    domainFields: List<DomainField>;
+}
+
+interface State {
+    loading: boolean;
+    ontologies: OntologyModel[];
+}
+
+export class OntologyLookupOptions extends PureComponent<Props, State> {
+    state: Readonly<State> = {
+        loading: true,
+        ontologies: undefined,
+    };
+
+    componentDidMount(): void {
+        this.loadData();
+    }
+
+    loadData = (): void => {
+        this.setState({
+            loading: true,
+        });
+
+        fetchOntologies(getServerContext().container.path).then(data => {
```

Comment:
I added a `catch` here to do a `console.error`. Since the ontologies state var is initialized to an empty array now, the dropdown will display as empty in this case, which makes sense. Added in a call to setState for loading: false.

Reply:
Looks good.